sys/auto_init/Makefile

```diff
@@ -14,4 +14,8 @@ ifneq (,$(filter auto_init_can,$(USEMODULE)))
   DIRS += can
 endif
 
+ifneq (,$(filter auto_init_gpio_exp,$(USEMODULE)))
+  DIRS += gpio_exp
+endif
```

Comment:
I'm not sure this belongs here. While it's necessary to add this into `DIRS`, it doesn't feel right because there's no auto_init code for this module... I'll check where it fits better.

Reply:
This is directly analogous to ```sys/auto_init/saul```. The directory is not for use by the API itself, but rather for implementations to add their auto_init code to. My first GPIO expander PR (#9054), is currently awaiting the merge of the I2C rework, and once it is done it would have auto_init code in this folder.

Reply:
I have removed this. Because the list of entries is not built dynamically like SAUL, drivers do not need auto_init code that is separate from the driver implementation.